### ISSUE TITLE
Dokkatoo uses typesafe Attributes, which are bugged in Gradle

### ISSUE THREAD

**Issue by aSemy** [adamko-dev/dokkatoo#214]
Gradle [recommends](https://docs.gradle.org/8.7/userguide/cross_project_publications.html#sec:variant-aware-sharing) using typesafe attributes to share files. However, [these are fundamentally bugged](https://github.com/gradle/gradle/issues/28695) and will not work in some situations.

I will implement a workaround, but I am opening this issue to track a real fix, dependent on whether Gradle can acknowledge and fix the issue.

This issue might be the cause of https://github.com/adamko-dev/dokkatoo/issues/165 and https://github.com/adamko-dev/dokkatoo/issues/170.

Thank you @kelvSYC for the report and reproducer! https://github.com/adamko-dev/dokkatoo/discussions/211

**Comment by CLOVIS-AI** [comment]
Do you know if this problem also applies to Dokka v2?

**Comment by aSemy** [comment]
> Do you know if this problem also applies to Dokka v2?

No, Dokkatoo and DGPv2 now use string-based attributes, which aren't affected.

**Comment by CLOVIS-AI** [comment]
Uh. That's strange, because I'm still seeing #170 with the latest Dokkatoo version.

Once I migrate to Dokka v2, I'll recreate the issue there.

**Comment by aSemy** [comment]
Ah yeah, #170 won't be helped by string-based attributes. A fix requires a more significant rewrite of how DGP shares files between subprojects.

I tried a workaround, but it doesn't work because of an (extremely confusing and undocumented) feature in Gradle https://github.com/gradle/gradle/issues/30921

#170 will be fixed in DGP at some point.

**Comment by aSemy** [comment]
Fixed by Gradle 8.14 https://github.com/gradle/gradle/pull/32277
